app/src/main/java/com/geeksville/mesh/repository/usb/UsbRepository.kt

```diff
@@ -0,0 +1,77 @@
+package com.geeksville.mesh.repository.usb
+
+import android.app.Application
+import android.hardware.usb.UsbDevice
+import android.hardware.usb.UsbManager
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.coroutineScope
+import com.geeksville.android.Logging
+import com.geeksville.mesh.CoroutineDispatchers
+import com.hoho.android.usbserial.driver.UsbSerialProber
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Repository responsible for maintaining and updating the state of USB connectivity.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+@Singleton
+class UsbRepository @Inject constructor(
```

Comment:
I want to add one more piece of functionality to this repository as a subsequent step: management of the serial communication itself, as currently handled by `SerialInterface`.  Doing this work on top of this PR's worth of code change was getting a little unwieldy so I stepped back and stabilized this portion of the change in isolation, first.